cmd/state-tray/internal/menu/projects.go

```diff
@@ -0,0 +1,73 @@
+package menu
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ActiveState/cli/cmd/state-tray/internal/open"
+	"github.com/ActiveState/cli/internal/graph"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/getlantern/systray"
+)
+
+type LocalProjectsUpdater struct {
+	menuItem *systray.MenuItem
+	items    []localProjectsMenuItem
+}
+
+type localProjectsMenuItem struct {
+	menuItem *systray.MenuItem
+	project  *graph.Project
+	cancel   context.CancelFunc
+}
+
+func NewLocalProjectsUpdater(menuItem *systray.MenuItem) *LocalProjectsUpdater {
+	return &LocalProjectsUpdater{menuItem, []localProjectsMenuItem{}}
+}
+
+func (u *LocalProjectsUpdater) Reload(projects []*graph.Project) {
+	for _, item := range u.items {
+		item.remove()
+	}
+
+	u.items = make([]localProjectsMenuItem, 0)
+	for _, project := range projects {
+		u.addLocalProject(project)
+	}
+}
```

Comment:
This is more heavy-handed than I would like. I tried to re-use the goroutines and `menuItems` however the `systray` library does not facilitate reordering the menu items. To ensure the order returned form the daemon is reflected in the menu we have to "remove" the original entries and create new ones. If this order is not important, it can be modified to re-use existing menu entries.